### PR TITLE
Add support for Matte layers

### PR DESCRIPTION
**Pull request by calda**
This PR adds support for layers with `MatteType.add`, which are used as the `mask` of the following layer.

| Before | After |
| ----- | ----- |
|   |  |

### CODE REVIEW

**Review by erichoracek — APPROVED** [review]
LGTM! Just two small suggestions

File: Sources/Private/Experimental/Layers/SublayerContainer.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Created by Cal Stephens on 1/27/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+import QuartzCore
+
+/// A base `CALayer` that manages the frame and animations
+/// of its `sublayers` and `mask`
+class BaseAnimationLayer: CALayer, AnimationLayer {
```
[the remark this answers is not in the document]
Comment (calda) [reply Sources/Private/Experimental/Layers/SublayerContainer.swift:8]: aha whoops, thanks

File: Sources/Private/Experimental/Layers/CALayer+setupLayerHierarchy.swift [hunk 2]
```diff
@@ -24,48 +24,102 @@ extension CALayer {
...
   }
 
 }
+
+extension Array where Element == LayerModel {
+  /// Pairs each `LayerModel` within this array with
+  /// a `LayerModel` to use as its mask, if applicable
+  /// based on the layer's `MatteType` configuration.
+  ///  - Assumes the layers are sorted in z-axis order.
+  var layersAndMasks: [(layer: LayerModel, mask: LayerModel?)] {
```
[the remark this answers is not in the document]
Comment (calda) [reply Sources/Private/Experimental/Layers/CALayer+setupLayerHierarchy.swift:97]: good idea

File: Sources/Private/Experimental/Layers/CALayer+setupLayerHierarchy.swift [hunk 3]
```diff
@@ -24,48 +24,102 @@ extension CALayer {
...
+
+      else {
+        addSublayer(parentTransformLayer)
       }
     }
   }
 
 }
+
+extension Array where Element == LayerModel {
```
[the remark this answers is not in the document]
Comment (calda) [reply Sources/Private/Experimental/Layers/CALayer+setupLayerHierarchy.swift:92]: Aha good idea! I was using `Int` indicies at first which required using `Array` (or at least didn't work with `ReversedCollection`) but now that it's using `removeFirst()` we can probably use `Collection` itself

[the remark this answers is not in the document]
Comment (calda) [reply Sources/Private/Experimental/Layers/CALayer+setupLayerHierarchy.swift:92]: Apparently `removeFirst()` is O(n), yeesh, probably don't want to be using this in a `while` loop then
> Calling this method may invalidate any existing indices for use with this collection.
Complexity: O(n), where n is the length of the collection.

**Pull request merged**